unittest/src/fake/temperature/mod.rs

```diff
@@ -0,0 +1,87 @@
+//! Fake implementation of the Temperature API, documented here:
+//! https://github.com/tock/tock/blob/master/doc/syscalls/60000_ambient_temperature.md
+//!
+//! Like the real API, `Temperature` controls a fake temperature sensor. It provides
+//! a function `set_value` used to immediately call an upcall with a temperature value read by the sensor
+//! and a function 'set_value_sync' used to call the upcall when the read command is received.
+
+use crate::{DriverInfo, DriverShareRef};
+use libtock_platform::{CommandReturn, ErrorCode};
+use std::cell::Cell;
+
+pub struct Temperature {
+    busy: Cell<bool>,
+    upcall_on_command: Cell<Option<i32>>,
```

Comment:
Please add a comment explaining why this is required, I assume others will have this issue.